rules/msk_topic_config_comments.go

```diff
@@ -0,0 +1,317 @@
+		return nil, err
+	}
+
+	// first look for the comment on the same line, after the property definition.
+	// Example: "retention.ms" = "2629800000" # keep data for 30 days
+	afterPropertyIdx := slices.IndexFunc(comments, func(comment hclsyntax.Token) bool {
+		return comment.Range.Start.Line == pair.Key.Range().Start.Line &&
+			comment.Range.Start.Column > pair.Value.Range().End.Column
+	})
+
+	if afterPropertyIdx >= 0 {
+		return &comments[afterPropertyIdx], nil
+	}
+
+	/* second, look for the comment on the previous line, before the property definition. Example:
+	# keep data for 30 days
+	"retention.ms" = "2629800000"
+	*/
+	beforePropertyIdx := slices.IndexFunc(comments, func(comment hclsyntax.Token) bool {
+		return comment.Range.Start.Line == pair.Key.Range().Start.Line-1 &&
+			comment.Range.End.Line == pair.Key.Range().Start.Line
+	})
+	if beforePropertyIdx >= 0 {
+		return &comments[beforePropertyIdx], nil
+	}
+
+	return nil, nil
+}
+
+func (r *MSKTopicConfigCommentsRule) getCommentsForFile(
+	runner tflint.Runner,
+	filename string,
+) (hclsyntax.Tokens, error) {
+	// we need to parse the file every time, otherwise keeping a cache per file doesn't work
+	file, err := runner.GetFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("getting hcl file %s for reading comments: %w", filename, err)
+	}
+
+	tokens, diags := hclsyntax.LexConfig(file.Bytes, filename, hcl.InitialPos)
+	if diags != nil {
+		return nil, diags
+	}
+
+	return slices.DeleteFunc(tokens, isNotComment), nil
+}
+
+func isNotComment(token hclsyntax.Token) bool {
+	return token.Type != hclsyntax.TokenComment
+}
+
+func (r *MSKTopicConfigCommentsRule) buildDurationComment(
+	runner tflint.Runner,
+	timePair hcl.KeyValuePair,
+	configValueInfo configTimeValueCommentInfo,
+) (string, error) {
+	var timeVal string
+	diags := gohcl.DecodeExpression(timePair.Value, nil, &timeVal)
+	if diags.HasErrors() {
+		return "", diags
+	}
+
+	if timeVal == configValueInfo.infiniteValue {
+		return fmt.Sprintf("# %s forever", configValueInfo.baseComment), nil
+	}
+
+	timeMillis, err := strconv.Atoi(timeVal)
+	if err != nil {
+		if configValueInfo.issueWhenInvalid {
+			issueMsg := fmt.Sprintf(
+				"%s must have a valid integer value expressed in milliseconds",
+				configValueInfo.key,
+			)
+			err := runner.EmitIssue(r, issueMsg, timePair.Value.Range())
+			if err != nil {
+				return "", fmt.Errorf("emitting issue: invalid time value: %w", err)
+			}
+		}
+
+		return "", nil
+	}
+
+	baseComment := configValueInfo.baseComment
+
+	msg := buildCommentForMillis(timeMillis, baseComment)
+	return msg, nil
+}
+
+func buildCommentForMillis(timeMillis int, baseComment string) string {
+	timeUnits, unit := determineTimeUnits(timeMillis)
+
+	msg := fmt.Sprintf("# %s for %d %s", baseComment, timeUnits, unit)
+	return msg
+}
+
+func determineTimeUnits(millis int) (int, string) {
+	// todo: this is not really perfect, as if the time is not exact in millis we'll output a partial number
```

Comment:
should this function return a `float` to handle fractional cases (e.g 1.5 months)? Is that what this comment is referring to?

Reply:
Indeed, that's the thing. I'll handle in another PR